src/meta/protos/proto/metadata.proto

```diff
@@ -28,6 +28,11 @@ message DataSchema {
 
   // Other optional table info
   map<string, string> metadata = 2;
+
+  uint32 max_column_id = 3;
```

Comment:
What about naming it `next_column_id`?

`max_column_id` looks like the max value of a valid column id